pelican/__init__.py

```diff
@@ -74,6 +74,9 @@ def init_plugins(self):
                 logger.error('Cannot register plugin `%s`\n%s',
                              plugin.__name__, e)
 
+        if 'PLUGINS' in self.settings and self.settings['PLUGINS'] is not None:
+            self.settings['PLUGINS'] = stringify_plugins(self.settings['PLUGINS'])
+
```

Comment:
I was thinking of a slightly different approach actually. We can simplify things. I'd say not bothering with `PLUGINS` setting at all here. `load_plugins` handles that and returns a list of objects that are plugin-like. All we need is a way to turn them into strings. I'd say scrapping `stringify_plugins` and changing `_stringify_plugin` to `get_plugin_name` with some modification. Code here would look like:

```python
    def init_plugins(self):
        self.plugins = []
        for plugin in load_plugins(self.settings):
            name = get_plugin_name(plugin)
            logger.debug('Registering plugin `%s`', name)
            try:
                plugin.register()
                self.plugins.append(plugin)
            except Exception as e:
                logger.error('Cannot register plugin `%s`\n%s',
                             name, e)
        self.settings['PLUGINS'] = [get_plugin_name(plugin) for plugin in self.plugins]
```

and `get_plugin_name` would look like:
```python
def get_plugin_name(plugin):
    try:
        return plugin.__class__.__qualname__
    except AttributeError:
        try:
            return plugin.__name__
         except AttributeError:
            pass
    return str(plugin)
```

PS: I changed `self.plugins` to store only registered plugins.

Reply:
Ah ok, now I get it. That makes a lot of sense.